lib/elixir/lib/code/formatter.ex

```diff
@@ -1263,15 +1267,15 @@ defmodule Code.Formatter do
     not Enum.any?(args, &match?({:<-, _, [_, _]}, &1))
   end
 
-  defp do_end_blocks([{{:__block__, meta, [:do]}, _} | rest] = blocks, state) do
-    if meta[:format] == :block or can_force_do_end_blocks?(rest, state) do
+  defp do_end_blocks(meta, [{{:__block__, _, [:do]}, _} | rest] = blocks, state) do
+    if is_list(meta[:do]) or can_force_do_end_blocks?(rest, state) do
```

Comment:
Nitpick: what do you think about moving this to a `meta?(meta, key)` function or similar?